open-apparel-registry/open-apparel-registry
# Add new facility details sidebar

## Overview

In order to allow maintenance work to be released while the Extended
Profile feature is in development, the new sidebar is wrapped in a
feature flag. When the flag is disabled, the old sidebar is displayed.

This initial implementation of the ExtendedProfile sidebar has only a
limited subset of fields and behaviors implemented. Specifically,
the claim status header; the OAR ID; the facility name, address, and
location; the static map; the contributors; and the action buttons have
been implemented.

Extended fields, claim fields, contributor/nonstandard fields, the
closed facility status and activity information, and the 'additional
information' popups have not yet been implemented.

Connects #1539 

## Demo
Unclaimed
<img width="175" alt="Screen Shot 2021-12-08 at 3 23 05 PM" src="https://user-images.githubusercontent.com/21046714/145283672-720755a1-4c64-48b9-8849-a654adefd5f5.png">
Pending
<img width="177" alt="Screen Shot 2021-12-08 at 3 22 49 PM" src="https://user-images.githubusercontent.com/21046714/145283697-e7cd8b22-5748-4cba-bc0d-3bce09a32597.png">
Embedded
<img width="236" alt="Screen Shot 2021-12-08 at 3 21 57 PM" src="https://user-images.githubusercontent.com/21046714/145283758-cde9cd3e-059b-4a00-b08e-d3a42122fa3f.png">
Claimed
<img width="1215" alt="Screen Shot 2021-12-08 at 4 03 36 PM" src="https://user-images.githubusercontent.com/21046714/145284246-42006bef-5a94-4b5a-82ac-5e9b93006c7b.png">
<img width="1434" alt="Screen Shot 2021-12-08 at 4 04 28 PM" src="https://user-images.githubusercontent.com/21046714/145284209-b94b8f79-9bbd-46fc-afd8-46e6c67e2b3e.png">
<img width="177" alt="Screen Shot 2021-12-08 at 4 04 43 PM" src="https://user-images.githubusercontent.com/21046714/145284293-11777953-fb94-47ff-8568-bb53bb58100d.png">


## Notes

We have followup cards for adding the 'additional information' popups and extended fields, and for adding the closed facility status and activity information. 

We should create an additional task to add contributor / nonstandard fields. 

The embedded map view of the sidebar wasn't presented in the wireframes. In the original embedded map view, we replace the action buttons with a link to the OAR; remove the contributors list; and hide 'other addresses', 'other locations', and so on. I've replicated this in the new sidebar. 

The header was a particular special case now that it shows claim information; based on previous requests from clients regarding the embedded map mode, clients prefer no information regarding other contributors to be displayed in their maps, and we also want no buttons to things like claims in embed mode. I've instead displayed 'contributed by (embedded map contributor)' in the header. 

There is a special state in the original sidebar where a contributor has claimed a facility but that claim is pending. During this state, other users can still access the 'claim this facility' button, but the pending claimant sees a 'pending claim' message. I've also replicated this in the new sidebar. 

## Testing Instructions

* Run `./scripts/update`
* View the details for a facility and confirm the sidebar behaves as before. 
* Turn on the `extended_profile` switch in the Django admin. Also give contributor 2 embedded map permissions. 
* Login as `c2@example.com` and view an unclaimed facility with multiple addresses/etc in the sidebar. Confirm that it matches [the wireframes](https://app.abstract.com/projects/441633f0-7d8c-11e9-af1e-972cbddca60a/branches/df31450b-d39a-4306-803d-c0d926ffd046/commits/72f5e69bf7a06d99182d72a68d99a26dfba4e1e8/files/fc1791e3-b8d2-4182-b240-fef32fabe978/layers/0EE5B838-BBA3-4D30-B62E-6127BF3E87B7?collectionId=1390af9f-d01b-4801-95a8-d8d14937a521&collectionLayerId=ffae49a3-4f5b-450a-bb3a-1926431af19e&mode=build) and that the data presented is correct. Confirm that the back button, contributor profile buttons, 'report a facility', and 'edit' buttons work as expected.
* View a facility with no 'other addresses' etc data and confirm the facility details do not display the additional information arrows. 
* View the multi-address facility in your embedded map preview. Adjust the color and font variables to confirm they work as expected. Confirm that the main header button and the list items are not clickable. 
* In regular view, claim one facility using the header button and one facility using the 'claim this facility' action button at the bottom of the sidebar. 
* As an admin user, approve one of the two claims.
* As `c2@example.com`, view the pending facility and confirm that the header shows a pending message and the 'claim this facility' action button is not displayed. 
* View the approved claim facility and confirm that the header shows the claim information and is not clickable (other than the back button). The 'claim this facility' and 'dispute claim' buttons shouldn't appear.
* As `c3@example.com`, view the pending facility - the claim buttons should be present. 
* As `c3@example.com`, view the approved claim - the 'dispute claim' button should be present and the header should show the claim information. 
* Mark a facility as having inexact_coordinates in the Django admin, then view the inexact coordinates message in full view and in embedded mode in the sidebar. 
* View the sidebar in mobile size and ensure it displays properly. 

## Checklist
